### ISSUE TITLE
[1.14.4] Block::getNameTextComponent called on server when it's @OnlyIn(Dist.CLIENT)

### ISSUE THREAD

**Issue by duely** [Direwolf20-MC/BuildingGadgets#410]
* Forge version 28.1.87
* Building Gadgets version 3.1.1b (provided to avoid the crash with another mod)
* *If the issue is a crash*; https://gist.github.com/duely/e8cd942b0a358539b256b5d48487701f
* How to reproduce the issue: The issue appears to happen when attempting to select the lower half of a dark oak door with the exchanging gadget

Based on MCP bot: ![](https://i.imgur.com/Th1IsTW.png)

As noted by it being client-side only, I double-checked my locally decompiled source:

```java
   @OnlyIn(Dist.CLIENT)
   public ITextComponent getNameTextComponent() {
      return new TranslationTextComponent(this.getTranslationKey());
   }```

I'm not sure why it's being called on the server in this case, but definitely something to consider.

**Comment by MichaelHillcox** [comment]
Ugh... Thanks, I'll take a look.

**Comment by duely** [comment]
Honestly, I have no idea why it's `@OnlyIn(Dist.CLIENT)`, considering "getTranslationKey" is a standard server-side method. Especially considering the fact that the majority of time you'd be wanting to reference the block by sending a text component from the key...

Especially considering all it does it return a new `TranslationTextComponent` -- there's nothing client-side only about it, I guess it was just only ever used on the client side code and when it was merged into the combined JAR it got annotated...

It might be worth submitting a patch to Forge to remove that.

**Comment by MichaelHillcox** [comment]
We're likely trying to call it from the server.

**Comment by duely** [comment]
Right, I understand, I was just saying: it *makes sense* to call it from the server; it doesn't make sense for it to be `@OnlyIn(Dist.CLIENT)`.

**Comment by MichaelHillcox** [comment]
Yeah, that is actually weird.
